rotkehlchen/chain/ethereum/modules/uniswap/v2/decoder.py

```diff
@@ -144,6 +146,42 @@ def _maybe_enrich_lp_tokens_transfers(
             lp_token_symbol=UNISWAP_PROTOCOL,
         )
 
+    def _fix_order_of_events_for_swaps(
+            self,
+            transaction: EvmTransaction,  # pylint: disable=unused-argument
+            decoded_events: list[HistoryBaseEntry],
+            all_logs: list[EvmTxReceiptLog],  # pylint: disable=unused-argument
+    ) -> list[HistoryBaseEntry]:
+        # look for the swap send & receive
+        send_event_idx, receive_event_idx = None, None
+        for idx, event in enumerate(decoded_events):
+            if (
+                event.event_type == HistoryEventType.TRADE and
+                event.event_subtype == HistoryEventSubType.SPEND and
+                event.counterparty == CPT_UNISWAP_V2
+            ):
+                send_event_idx = idx
+
+            elif (
+                event.event_type == HistoryEventType.TRADE and
+                event.event_subtype == HistoryEventSubType.RECEIVE and
+                event.counterparty == CPT_UNISWAP_V2
+            ):
+                receive_event_idx = idx
+
+        if (
+            send_event_idx is not None and
+            receive_event_idx is not None and
+            receive_event_idx != send_event_idx + 1
+        ):
+            maybe_reshuffle_events(
+                out_event=decoded_events[send_event_idx],
+                in_event=decoded_events[receive_event_idx],
+                events_list=decoded_events,
+            )
```

Comment:
This looks to me that could happen in other cases too. You have the transaction, is it using a router or something special? if that is the case maybe uniswap v3 could also have this

Reply:
It was via the router.

Reply:
if is a thing of the router isn't possible to filter the transactions for which this function runs using the router address? We have `UNISWAP_ROUTERS`... isn't possible to do

```
if transaction.to in UNISWAP_ROUTERS:
    return
```

?

This will prevent the logic from executing for every transaction

Reply:
`UNISWAP_ROUTERS` is for AUTO routers but this is via normal uniswap v2 routers